### PR TITLE
fix datarace in defaultFontCache

### LINKED ISSUE #413: defaultFontCache is not thread safe
Discovered a data race while rendering plots from a web server using multiple goroutines.

PR coming soon

### PR DESCRIPTION
**Pull request by campoy**
Fixes #413

### CODE REVIEW

File: vg/vgimg/vgimg.go [hunk 1]
```diff
@@ -297,11 +300,12 @@ func (c *Canvas) DrawImage(rect vg.Rectangle, img image.Image) {
...
-var (
-	// RegisteredFont contains the set of font names
-	// that have already been registered with draw2d.
-	registeredFont = map[string]bool{}
-)
+// registeredFont contains the set of font names
+// that have already been registered with draw2d.
+var registeredFont = struct {
+	m map[string]bool
+	sync.Mutex
```
Comment (kortschak) [inline vg/vgimg/vgimg.go]: Please place the lock above the map.

Comment (campoy) [reply vg/vgimg/vgimg.go]: done

**Comment by campoy** [comment]
What do you mean by A+C?

File: vg/vgimg/vgimg_test.go [hunk 2]
```diff
@@ -48,3 +49,20 @@ func TestIssue179(t *testing.T) {
...
+		c := vgimg.New(215, 215)
+		c.FillString(vg.Font{Size: 10}, vg.Point{}, "hi")
+		wg.Done()
+	}()
+	go func() {
+		c := vgimg.New(215, 215)
+		c.FillString(vg.Font{Size: 10}, vg.Point{}, "hi")
+		wg.Done()
+	}()
+	wg.Done()
```
Comment (sbinet) [inline vg/vgimg/vgimg_test.go]: Shouldn't this read 'wg.Wait()' instead?

Comment (campoy) [reply vg/vgimg/vgimg_test.go]: duh!

**Comment by kortschak** [comment]
Thanks

**Pull request merged**